mitosheet/mitosheet/steps_manager.py

```diff
@@ -395,7 +397,10 @@ def handle_edit_event(self, edit_event: Dict[str, Any]) -> None:
 
         step_performer = EVENT_TYPE_TO_STEP_PERFORMER[edit_event["type"]]
 
-        # First, we make a new step
+        # First, we add the public interface to the params, as we might need it for any step
+        edit_event["params"]['public_interface_version'] = self.public_interface_version
```

Comment:
Instead of putting this here and then having to upgrade a bunch of steps to handle it as a new param, we could just put it in the state since its for the entire analysis and not just individual steps.

Reply:
This is what I did initially. Things got really messy, and I refactored to this. I think putting here is fine - since we will know to just add it to new steps if we need it from now on!